src/base/bittorrent/session.cpp

```diff
@@ -1415,85 +1419,52 @@ void Session::configure(libtorrent::settings_pack &settingsPack)
-    // so instead we use address_v4::broadcast()
-    // Proactively do the same for 0.0.0.0 and address_v4::any()
-    f.add_rule(libt::address_v4::any()
-               , libt::address_v4::broadcast()
-               , 1 << libt::session::global_peer_class_id);
-#if TORRENT_USE_IPV6
-    // IPv6 may not be available on OS and the parsing
-    // would result in an exception -> abnormal program termination
-    // Affects Windows XP
-    try {
-        f.add_rule(libt::address_v6::from_string("::0")
-                   , libt::address_v6::from_string("ffff:ffff:ffff:ffff:ffff:ffff:ffff:ffff")
-                   , 1 << libt::session::global_peer_class_id);
-    }
-    catch (std::exception &) {}
-#endif
-    if (ignoreLimitsOnLAN()) {
-        // local networks
-        f.add_rule(libt::address_v4::from_string("10.0.0.0")
-                   , libt::address_v4::from_string("10.255.255.255")
-                   , 1 << libt::session::local_peer_class_id);
-        f.add_rule(libt::address_v4::from_string("172.16.0.0")
-                   , libt::address_v4::from_string("172.31.255.255")
-                   , 1 << libt::session::local_peer_class_id);
-        f.add_rule(libt::address_v4::from_string("192.168.0.0")
-                   , libt::address_v4::from_string("192.168.255.255")
-                   , 1 << libt::session::local_peer_class_id);
-        // link local
-        f.add_rule(libt::address_v4::from_string("169.254.0.0")
-                   , libt::address_v4::from_string("169.254.255.255")
-                   , 1 << libt::session::local_peer_class_id);
-        // loopback
-        f.add_rule(libt::address_v4::from_string("127.0.0.0")
-                   , libt::address_v4::from_string("127.255.255.255")
-                   , 1 << libt::session::local_peer_class_id);
-#if TORRENT_USE_IPV6
-        // IPv6 may not be available on OS and the parsing
-        // would result in an exception -> abnormal program termination
-        // Affects Windows XP
-        try {
-            // link local
-            f.add_rule(libt::address_v6::from_string("fe80::")
-                       , libt::address_v6::from_string("febf:ffff:ffff:ffff:ffff:ffff:ffff:ffff")
-                       , 1 << libt::session::local_peer_class_id);
-            // unique local addresses
-            f.add_rule(libt::address_v6::from_string("fc00::")
-                       , libt::address_v6::from_string("fdff:ffff:ffff:ffff:ffff:ffff:ffff:ffff")
-                       , 1 << libt::session::local_peer_class_id);
-            // loopback
-            f.add_rule(libt::address_v6::from_string("::1")
-                       , libt::address_v6::from_string("::1")
-                       , 1 << libt::session::local_peer_class_id);
-        }
-        catch (std::exception &) {}
-#endif
-    }
-    m_nativeSession->set_peer_class_filter(f);
+    // Assuming that libtorrent will not give 0 as class ID
+    if (m_UTPClassID != 0) return;
```

Comment:
This method is called only once, isn't it? Then why this check?

Reply:
Overzealous checking in case someone else in the future tries to call it again. I'll convert to Q_ASSERT.